content/docs/05-ai-sdk-ui/50-stream-protocol.mdx

```diff
@@ -0,0 +1,195 @@
+---
+title: Stream Protocols
+description: Learn more about the supported stream protocols in the Vercel AI SDK.
+---
+
+# Stream Protocols
+
+AI SDK UI functions such as `useChat` and `useCompletion` support both text streams and data streams.
+The stream protocol defines how the data is streamed to the frontend on top of the HTTP protocol.
+
+This page describes both protocols and how to use them in the backend and frontend.
+
+You can use this information develop custom backends and frontends for your use case, e.g.,
+to provide compatible API endpoints that are implemented in a different language such as Python.
+
+## Text Stream Protocol
+
+Text streams are simple text chunks that are streamed to the frontend.
```

Comment:
"A text stream contains chunks in plain text, that are streamed to the frontend. Each chunk is then appended together to form a full text response."